src/Fixer/ReturnNotation/ReturnAssignmentFixer.php

```diff
@@ -0,0 +1,349 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ReturnNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class ReturnAssignmentFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Non global, static or indirect referenced variables should not be assigned and directly returned by a function or method.',
+            [new CodeSample("<?php\nfunction a() {\n    \$a = 1;\n    return \$a;\n}\n")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // FIXME
+        // should be run before white space lines, extra empty lines,
```

Comment:
and before `blank_line_before_return`/`blank_line_before_statement`

Reply:
I don't think that is needed, the fixer removes the line where the statement was on if the line is empty

Reply:
it's not about removing, it's about adding new line.

```php
function foo()
{
    $foo = [1, 2, 3];
    $bar = array_filter($foo, ...);

    return $bar;
}
```

what would be the outcome?

```php
function foo()
{
    $foo = [1, 2, 3];
    return array_filter($foo, ...);
}
```

so far, so good, right?

but, `blank_line_before_return` want to put blank line before `return`.
if it will be executed on original code, it will say that all is fine.
if it will be executed on fixed code, it will add new line.

for that, `return_assignment` shall be executed before `blank_line_before_return`